### PR TITLE
Promtheus: Improve tab completion

### PR DESCRIPTION
**Pull request by davkal**
Change in behavior:

- no longer suggest everything in empty field, only history
- term suggestions need at least one character to suggest metric or function

Somehow multi-line suggestions are not working much. But that's not a regression in this PR.

### CODE REVIEW

**Review by pracucci — COMMENTED** [review]
I'm not familiar with Grafana internals, so I can't comment just reviewing the code. I've built it and tried it. I think there's some problem with the terms completion and digits, ie:

**Comment by davkal** [comment]
The issue with numbers in the recording rule name is not a regression though.

File: public/app/plugins/datasource/prometheus/language_provider.test.ts [hunk 1]
```diff
@@ -101,14 +87,31 @@ describe('Language completion provider', () => {
   });
 
   describe('metric suggestions', () => {
-    it('returns metrics and function suggestions in an unknown context', async () => {
+    it('returns history, metrics and function suggestions in an uknown context ', async () => {
```
Comment (pracucci) [inline public/app/plugins/datasource/prometheus/language_provider.test.ts]: `uknown` > `unknown`

**Review by pracucci — APPROVED** [review]
Manually trying it - and excluding other issues which look not being a regression of this PR - the changes in this PR look doing what the PR description says :)

**Review by kaydelaney — APPROVED** [review]
LGTM

**Pull request merged**